test/integration/test_phone_number_lookup.py

```diff
@@ -0,0 +1,168 @@
+
+    
+    def testFailedPhoneNumberLookup(self):
+        """Test Phone Number Lookup API with bad data to force an error"""
+        with self.assertRaises(bandwidth.ApiException) as context:
+            lookup_request = LookupRequest(
+                tns=[
+                    'not a number',
+                ],
+            )
+            self.api_instance.create_lookup(self.account_id, lookup_request)
+        
+        self.assertIs(type(context.exception.status), int)
+        self.assertIs(type(context.exception.body), str)
+
+        # initialize TnLookupRequestError model 
+        error = TnLookupRequestError(message=(json.loads(context.exception.body))['message'])
+        self.assertIs(type(error), TnLookupRequestError)
+
+    
+    def testUnauthorizedRequest(self):
+        configuration = bandwidth.Configuration(
+            username = 'bad_username',
+            password = 'bad_password'
+        )
+        unauthorized_api_client = bandwidth.ApiClient(configuration)
+        unauthorized_api_instance = phone_number_lookup_api.PhoneNumberLookupApi(unauthorized_api_client)
+        lookup_request = LookupRequest(
+            tns=[
+                os.environ['BW_NUMBER']
+            ],
+        )
+
+        with self.assertRaises(UnauthorizedException) as context:
+            unauthorized_api_instance.create_lookup(self.account_id, lookup_request)
+        
+        self.assertIs(type(context.exception), UnauthorizedException)
+        self.assertIs(type(context.exception.status), int)
+        self.assertEqual(context.exception.status, 401)
+        self.assertIs(type(context.exception.body), str)
+    
+    def testForbiddenRequest(self):
+        configuration = bandwidth.Configuration(
+            username = os.environ['BW_USERNAME_FORBIDDEN'],
+            password = os.environ['BW_PASSWORD_FORBIDDEN']
+        )
+        forbidden_api_client = bandwidth.ApiClient(configuration)
+        forbidden_api_instance = phone_number_lookup_api.PhoneNumberLookupApi(forbidden_api_client)
+        lookup_request = LookupRequest(
+            tns=[
+                os.environ['BW_NUMBER']
+            ],
+        )
+
+        # This API throws a 401 when a user provides valid credentials with the `TN Lookup` role disabled
+        # with self.assertRaises(ForbiddenException) as context:
+        with self.assertRaises(UnauthorizedException) as context:
+            forbidden_api_instance.create_lookup(self.account_id, lookup_request)
+        
+        # self.assertIs(type(context.exception), ForbiddenException)
+        # self.assertEqual(context.exception.status, 403)
+        self.assertIs(type(context.exception), UnauthorizedException)
+        self.assertIs(type(context.exception.status), int)
+        self.assertEqual(context.exception.status, 401)
+        self.assertIs(type(context.exception.body), str)
```

Comment:
Generally speaking look for opportunities to remove duplicate code.

Reply:
Agreed - wasnt sure if we wanted to leave this in here to test the current behavior that _should_ cause a 403 from the server - could write a helper function to test bad auth with an exception and status code parameter though and then just update the parameters when the incorrect 401 changes to a 403

Reply:
Implemented  a `validateAuthException` function to address this